db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SubqueryNode.java

```diff
@@ -2702,4 +2699,11 @@ public int getCardinalityCheck() throws StandardException {
             return OnceResultSet.DO_CARDINALITY_CHECK;
         }
     }
+
+    @Override
+    public ValueNode replaceIndexExpression(ResultColumnList childRCL) throws StandardException {
+        assert resultSet instanceof Optimizable;
```

Comment:
resultSet of a SubqueryNode could be a SelectNode, but when we get it, it is called from modifyAccessPath phase, so I guess SelectNode has already been replaced by a ProjectRestrictNode, so the assertion here is OK. Am I right? A comment may be helpful here.

Reply:
I think that's right. When we enter this call, it should come from the `replaceIndexExpression` in `genProjectRestrict`. At that point, access paths of subqueries, including derived tables, have been modified. The other place index expressions got replaced is in `FromBaseTable`. But it should never ends up here.

I add a comment as you suggested.